docs/clients/promtail/stages/template.md

```diff
@@ -58,6 +58,17 @@ converts its value to be all lowercase. For example, if the extracted map
 contained `app` with a value of `LOKI`, this pipeline would change its value to
 `loki`.
 
+```yaml
+- template:
+    source: output_msg
```

Comment:
There is something weird here that should at least be documented.

If the source already exist, only `.Value` with the source value will be expanded in the template.

May be if the source exist we should add the `Value` from the source into the template data all the time to be more clear.

Reply:
Yes, if source exists, only `.Value` will be expanded. Will document it

Reply:
I actually prefer option 2, transform the extracted into template data, if source exist add it as `Value` key.

Reply:
Please also update the documentation to reflect that all keys are automatically available to be expanded + the extra source if available as `.Value`.

Reply:
Sure. Will do

Reply:
Have updated the doc. I still doubt if the statements I added make things clear. Give me your feedback once you read.